### ISSUE TITLE
Navigator Error - Type Error

### ISSUE THREAD

**Issue by ironfrown** [ContinuumIO/anaconda-issues#1837]
Anaconda navigator either returns `"the JSON object must be str, not 'bytes'"` or crashes with this traceback:
```
Traceback (most recent call last):
  File "/home/jacob/Apps/anaconda3/lib/python3.5/site-packages/anaconda_navigator/exceptions.py", line 75, in exception_handler
    return_value = func(*args, **kwargs)
  File "/home/jacob/Apps/anaconda3/lib/python3.5/site-packages/anaconda_navigator/app/start.py", line 115, in start_app
    window = run_app(splash)
  File "/home/jacob/Apps/anaconda3/lib/python3.5/site-packages/anaconda_navigator/app/start.py", line 58, in run_app
    window = MainWindow(splash=splash)
  File "/home/jacob/Apps/anaconda3/lib/python3.5/site-packages/anaconda_navigator/widgets/main_window.py", line 160, in __init__
    self.api = AnacondaAPI()
  File "/home/jacob/Apps/anaconda3/lib/python3.5/site-packages/anaconda_navigator/api/anaconda_api.py", line 1205, in AnacondaAPI
    ANACONDA_API = _AnacondaAPI()
  File "/home/jacob/Apps/anaconda3/lib/python3.5/site-packages/anaconda_navigator/api/anaconda_api.py", line 65, in __init__
    self._conda_api = CondaAPI()
  File "/home/jacob/Apps/anaconda3/lib/python3.5/site-packages/anaconda_navigator/api/conda_api.py", line 1622, in CondaAPI
    CONDA_API = _CondaAPI()
  File "/home/jacob/Apps/anaconda3/lib/python3.5/site-packages/anaconda_navigator/api/conda_api.py", line 340, in __init__
    self.set_conda_prefix()
  File "/home/jacob/Apps/anaconda3/lib/python3.5/site-packages/anaconda_navigator/api/conda_api.py", line 489, in set_conda_prefix
    self.ROOT_PREFIX = info['root_prefix']
TypeError: byte indices must be integers or slices, not str
```
System information:
```
               platform : linux-64
          conda version : 4.2.13
       conda is private : False
      conda-env version : 4.2.13
    conda-build version : 1.21.3
         python version : 3.5.3.final.0
       requests version : 2.14.2
       root environment : /home/jacob/Apps/anaconda3  (writable)
    default environment : /home/jacob/Apps/anaconda3
       envs directories : /home/jacob/Apps/anaconda3/envs
          package cache : /home/jacob/Apps/anaconda3/pkgs
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/anaconda-fusion/linux-64
                          https://conda.anaconda.org/anaconda-fusion/noarch
                          https://repo.continuum.io/pkgs/free/linux-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/pro/linux-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : /home/jacob/.condarc
           offline mode : False

```

**Comment by goanpeca** [comment]
@ironfrown please update to the latest Anaconda Navigator (1.6.2)

And make sure that you added anaconda to your bash profile. This erros occurs because conda is not found on the system path.

**Comment by tenberg** [comment]
I am running the latest version of AN (1.6.2). My .bash_profile has the following line and I am still getting the error:

export PATH="/Users/ed/anaconda/bin:$PATH"

**Comment by goanpeca** [comment]
@tenberg if you open the terminal and type 

`$ source activate root` and then

`$ anaconda-navigator`, what do you get?

**Comment by tenberg** [comment]
I get the same error:

[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0

**Comment by goanpeca** [comment]
Closing as duplicate of #1837

**Comment by chaseplastic** [comment]
Where precisely is the solution to this error? I see a lot of closed issues and cannot find a solution to the recurring problem.

**Comment by FeroEQ** [comment]
Do not close !!! This error occurs on fresh installations again and again !!!

**Comment by iainism** [comment]
Freshly downloaded and installed, using recommended method and settings.  First time to run:  Get an an appcrash

Try to use the built-in bug-repoorting: Get a "page not found" error

Manage to hack the URI to file a bug:  Get my report auto-closed, and a suggestion to update _something that I only just downloaded_

Try to update:  Fails.  "pythonw not found, attempting to roll back changes"

Thanks for the fantastic first time user experience.  I'm off to try ActivePython, perhaps that will actually, you know, work?

**Comment by iamnoob246** [comment]
had the exact same issue, tried updating via terminal and still no luck :>

**Comment by MsToT** [comment]
Ran into the same issue yesterday. Anaconda (prompt and navigator) are in path and have been working before. Spent more time looking in the Script folder and found that the following files have been change:
activate.bat >> activate.bat.c~

**Comment by vishu133** [comment]
my antivirus was blocking python. Unblocked it now it is working

**Comment by TheSirion** [comment]
I'm having the same problem. I reinstalled Anaconda twice now and the problem remains. I'm using Windows 10 64-bit. 

`Main Error

byte indices must be integers or slices, not str
Traceback

Traceback (most recent call last):
  File "C:\PROGRAMDATA\ANACONDA3\lib\site-packages\anaconda_navigator\exceptions.py", line 75, in exception_handler
    return_value = func(*args, **kwargs)
  File "C:\PROGRAMDATA\ANACONDA3\lib\site-packages\anaconda_navigator\app\start.py", line 115, in start_app
    window = run_app(splash)
  File "C:\PROGRAMDATA\ANACONDA3\lib\site-packages\anaconda_navigator\app\start.py", line 58, in run_app
    window = MainWindow(splash=splash)
  File "C:\PROGRAMDATA\ANACONDA3\lib\site-packages\anaconda_navigator\widgets\main_window.py", line 160, in __init__
    self.api = AnacondaAPI()
  File "C:\PROGRAMDATA\ANACONDA3\lib\site-packages\anaconda_navigator\api\anaconda_api.py", line 1205, in AnacondaAPI
    ANACONDA_API = _AnacondaAPI()
  File "C:\PROGRAMDATA\ANACONDA3\lib\site-packages\anaconda_navigator\api\anaconda_api.py", line 65, in __init__
    self._conda_api = CondaAPI()
  File "C:\PROGRAMDATA\ANACONDA3\lib\site-packages\anaconda_navigator\api\conda_api.py", line 1622, in CondaAPI
    CONDA_API = _CondaAPI()
  File "C:\PROGRAMDATA\ANACONDA3\lib\site-packages\anaconda_navigator\api\conda_api.py", line 340, in __init__
    self.set_conda_prefix()
  File "C:\PROGRAMDATA\ANACONDA3\lib\site-packages\anaconda_navigator\api\conda_api.py", line 489, in set_conda_prefix
    self.ROOT_PREFIX = info['root_prefix']
TypeError: byte indices must be integers or slices, not str`

I've tried the suggested fixes. No commands on the Anaconda prompt seem to work. I get this message every time:

`(C:\PROGRAMDATA\ANACONDA3) C:\Users\Matheus Ribeiro>conda update anaconda-navigator
usage: CONDA-script.py [-h] [-V] command ...
CONDA-script.py: error: argument command: invalid choice: 'C:\\PROGRAMDATA\\ANACONDA3\\SCRIPTS\\CONDA-script.py' (choose from 'info', 'help', 'list', 'search', 'create', 'install', 'update', 'upgrade', 'remove', 'uninstall', 'config', 'clean', 'package')`

**Comment by jasperdebeer** [comment]
Had the same problem on windows 10 64bit. Could not even use conda commands.
Managed to solve it by cleaning the windows registry using ccleaner

**Comment by TomNett** [comment]
when i open Conda command prompt i get this error so i cannot even do upgrade "C:\Anaconda3\Scripts\\..\Scripts\conda.exe"' is not recognized as an internal or external command,
operable program or batch file.

Any ideas?

**Comment by georgknabl** [comment]
Had the same issue on Windows. Updating packages did not solve it. I didn't want to set the PATH.

**My solution** was to manually enter the paths instead of using the "info[...]" references in set_conda_prefix in file \Anaconda3\lib\site-packages\anaconda_navigator\api\conda_api.py.

A better solution, of course, would be a more error tolerant approach facilitating a fallback by the developers in the first place but it does solve the problem quickly. Just make sure to backup the changes as they might get overridden after updating the package.

Since Anaconda is based mostly on viewable and editable python code, most problems can fortunately debugged and solved by oneself.

**Comment by mas-dse-greina** [comment]
Getting the same error. All of the links here seem to reference each other, but I don't see the actual solution. I have installed and re-install miniconda several times now. Creating a virtual environment doesn't work.


[buer@param ~]$ conda update conda
Fetching package metadata .........
Solving package specifications: .

Package plan for installation in environment /home/buer/miniconda2:

The following packages will be SUPERSEDED by a higher-priority channel:

    conda: 4.4.10-py27_0 https://repo.continuum.io/pkgs/main --> 4.3.30-py27h6ae6dc7_0

Proceed ([y]/n)? y

conda-4.3.30-p 100% |###############################################################################################| Time: 0:00:00   2.47 MB/s
usage: conda [-h] [-V] command ...
conda: error: argument command: invalid choice: '/home/buer/miniconda2/bin/conda' (choose from u'info', u'help', u'list', u'search', u'create', u'install', u'update', u'upgrade', u'remove', u'uninstall', u'config', u'clean', u'package')

**Comment by Serg132003** [comment]
have this navigator error on my windows:
Navigator Error
An unexpected error occurred on Navigator start-up

Report
Please report this issue in the anaconda issue tracker

Main Error
byte indices must be integers or slices, not str
Traceback
Traceback (most recent call last):
  File "C:\Users\serge\AppData\Local\Continuum\anaconda2\lib\site-packages\anaconda_navigator\exceptions.py", line 75, in exception_handler
    return_value = func(*args, **kwargs)
  File "C:\Users\serge\AppData\Local\Continuum\anaconda2\lib\site-packages\anaconda_navigator\app\start.py", line 120, in start_app
    window = run_app(splash)
  File "C:\Users\serge\AppData\Local\Continuum\anaconda2\lib\site-packages\anaconda_navigator\app\start.py", line 63, in run_app
    window = MainWindow(splash=splash)
  File "C:\Users\serge\AppData\Local\Continuum\anaconda2\lib\site-packages\anaconda_navigator\widgets\main_window.py", line 163, in __init__
    self.api = AnacondaAPI()
  File "C:\Users\serge\AppData\Local\Continuum\anaconda2\lib\site-packages\anaconda_navigator\api\anaconda_api.py", line 2004, in AnacondaAPI
    ANACONDA_API = _AnacondaAPI()
  File "C:\Users\serge\AppData\Local\Continuum\anaconda2\lib\site-packages\anaconda_navigator\api\anaconda_api.py", line 88, in __init__
    self._conda_api = CondaAPI()
  File "C:\Users\serge\AppData\Local\Continuum\anaconda2\lib\site-packages\anaconda_navigator\api\conda_api.py", line 1690, in CondaAPI
    CONDA_API = _CondaAPI()
  File "C:\Users\serge\AppData\Local\Continuum\anaconda2\lib\site-packages\anaconda_navigator\api\conda_api.py", line 347, in __init__
    self.set_conda_prefix()
  File "C:\Users\serge\AppData\Local\Continuum\anaconda2\lib\site-packages\anaconda_navigator\api\conda_api.py", line 496, in set_conda_prefix
    self.ROOT_PREFIX = info['root_prefix']
TypeError: byte indices must be integers or slices, not str

what to do?

**Comment by madhu-korada** [comment]
getting this and not able to open ananconda navigator
i did update conda

Traceback (most recent call last):
  File "C:\Users\madhu\Anaconda3\etc\keras\load_config.py", line 14, in <module>
    backend = json.load(fh).get("backend", backend)
  File "C:\Users\madhu\Anaconda3\lib\json\__init__.py", line 299, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "C:\Users\madhu\Anaconda3\lib\json\__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "C:\Users\madhu\Anaconda3\lib\json\decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Users\madhu\Anaconda3\lib\json\decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 6 column 5 (char 123)

(base) C:\Users\madhu>set /p KERAS_BACKEND= 0<temp.txt

(base) C:\Users\madhu>del temp.txt

(base) C:\Users\madhu>python -c "import keras"  1>nul 2>&1

(base) C:\Users\madhu>if errorlevel 1 (
ver  1>nul
 set "KERAS_BACKEND=theano"
 python -c "import keras"  1>nul 2>&1
)

**Comment by MichlF** [comment]
Updated Anaconda and get this error (Windows 10). Maybe related to me using PyCharm to update modules?

Navigator Error
An unexpected error occurred on Navigator start-up

Report
Please report this issue in the anaconda issue tracker

Main Error
could not determine a constructor for the tag 'tag:yaml.org,2002:python/unicode'
  in "C:\Users\Michl-Schlaptop\AppData\Local\ContinuumIO\binstar\config.yaml", line 1, column 1
Traceback
Traceback (most recent call last):
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\anaconda_navigator\exceptions.py", line 75, in exception_handler
    return_value = func(*args, **kwargs)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\anaconda_navigator\app\start.py", line 120, in start_app
    window = run_app(splash)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\anaconda_navigator\app\start.py", line 63, in run_app
    window = MainWindow(splash=splash)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\anaconda_navigator\widgets\main_window.py", line 163, in __init__
    self.api = AnacondaAPI()
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\anaconda_navigator\api\anaconda_api.py", line 2004, in AnacondaAPI
    ANACONDA_API = _AnacondaAPI()
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\anaconda_navigator\api\anaconda_api.py", line 89, in __init__
    self._client_api = ClientAPI()
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\anaconda_navigator\api\client_api.py", line 612, in ClientAPI
    CLIENT_API = _ClientAPI()
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\anaconda_navigator\api\client_api.py", line 82, in __init__
    log_level=logging.NOTSET
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\binstar_client\utils\config.py", line 96, in get_server_api
    config = get_config(remote_site=site)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\binstar_client\utils\config.py", line 229, in get_config
    file_configs = load_file_configs(SEARCH_PATH)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\binstar_client\utils\config.py", line 223, in load_file_configs
    raw_data = collections.OrderedDict(kv for kv in itertools.chain.from_iterable(load_paths))
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\binstar_client\utils\config.py", line 223, in 
    raw_data = collections.OrderedDict(kv for kv in itertools.chain.from_iterable(load_paths))
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\binstar_client\utils\config.py", line 203, in _dir_yaml_loader
    yield filepath, load_config(filepath)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\binstar_client\utils\config.py", line 187, in load_config
    data = yaml_load(fd)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\binstar_client\utils\yaml.py", line 9, in yaml_load
    return safe_load(stream)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\yaml\__init__.py", line 94, in safe_load
    return load(stream, SafeLoader)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\yaml\__init__.py", line 72, in load
    return loader.get_single_data()
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\yaml\constructor.py", line 37, in get_single_data
    return self.construct_document(node)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\yaml\constructor.py", line 46, in construct_document
    for dummy in generator:
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\yaml\constructor.py", line 398, in construct_yaml_map
    value = self.construct_mapping(node)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\yaml\constructor.py", line 204, in construct_mapping
    return super().construct_mapping(node, deep=deep)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\yaml\constructor.py", line 125, in construct_mapping
    key = self.construct_object(key_node, deep=deep)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\yaml\constructor.py", line 86, in construct_object
    data = constructor(self, node)
  File "C:\Users\Michl-Schlaptop\Anaconda3\lib\site-packages\yaml\constructor.py", line 414, in construct_undefined
    node.start_mark)
yaml.constructor.ConstructorError: could not determine a constructor for the tag 'tag:yaml.org,2002:python/unicode'
  in "C:\Users\Michl-Schlaptop\AppData\Local\ContinuumIO\binstar\config.yaml", line 1, column 1

**Comment by odgrigor** [comment]
@goanpeca Everything is referenced back to this main thread, but is there a solution to these issues?

Matter of fact, has anyone here figured it out yet?

**Comment by mahjoobrad** [comment]
> I was having the same error. The following commands worked for me:
> conda update conda
> conda update --all

@zahranoorcs 
I tried these codes but unfortunately nothing has changed.
ConfigurationLoadError: Unable to load configuration file.
Path: C:\Users\2016\.condarc
reason: invalid yaml at position 0
